src/hypofuzz/utils.py

```diff
@@ -51,3 +53,30 @@ def value(self, value: T) -> None:
 
 def lerp(a: float, b: float, t: float) -> float:
     return (1 - t) * a + t * b
+
+
+def k_way_merge(
+    lists: Sequence[Sequence[T]], key: Callable[[T], Any] = lambda x: x
+) -> list[T]:
+    # merges k sorted lists in O(nlg(k)) time, where n is the total number of
+    # elements.
+    #
+    # NOTE: this implementation is *not* a stable sort, since the heap key
+    #   (key(l[0]), i, 0)
+    # falls back to i when key(l[0]) is equal.
+    result: list[T] = []
+    heap: list[tuple[Any, int, int]] = []
+    assert all(len(l) > 0 for l in lists)
```

Comment:
Instead of asserting that the lists are nonempty, let's just `lists = [ls for ls in lists if ls]`